sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/BigDecimalConverter.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.util
+
+trait BigDecimalConverter[T] {
+  def toBigDecimal(in: T): BigDecimal
+  def fromBigDecimal(bd: BigDecimal): T
+}
+
+/**
+ * Helper type converters to work with BigDecimal
+ * from http://stackoverflow.com/a/30979266/1115193
+ */
+object BigDecimalConverter {
```

Comment:
And also, I've found #6836 also use BigDecimal.setScale to do round. when it comes to deal more dataTypes, rather than just Decimal Float and Double, I would prefer to generalize like this.